PyInstaller/compat.py

```diff
@@ -72,8 +73,8 @@
 # disabling the compatibility mode and using python that does not properly support Big Sur still leaves find_library()
 # broken (which is a scenario that we ignore at the moment).
 # The same logic applies to macOS 12 (Monterey).
-is_macos_11_compat = _macos_ver and _macos_ver[0:2] == (10, 16)  # Big Sur or newer in compat mode
-is_macos_11_native = _macos_ver and _macos_ver[0:2] >= (11, 0)  # Big Sur or newer in native mode
+is_macos_11_compat = bool(_macos_ver and _macos_ver[0:2] == (10, 16))  # Big Sur or newer in compat mode
+is_macos_11_native = bool(_macos_ver and _macos_ver[0:2] >= (11, 0))  # Big Sur or newer in native mode
```

Comment:
The `bool()` needs to be explicitly applied only to the first part of the statement, where we check `_macos_ver` for truthy-ness. The second of part the statement is already a bool due to operator type.